adam/curriculum/imprecise_descriptions_curriculum.py

```diff
@@ -305,6 +314,139 @@ def make_jump_imprecise_temporal_descriptions(
     )
 
 
+def make_take_grab_subtle_verb_distinction(
+    num_samples: int = 5, *, num_noise_objects: int = 0  # pylint:disable=unused-argument
+) -> Phase1InstanceGroup:
+    taker = standard_object("tosser_passer_0", THING, required_properties=[ANIMATE])
+    takee = standard_object("tossee_passee_0", THING, required_properties=[INANIMATE])
+    return phase1_instances(
+        "taking-grabbing",
+        chain(
+            flatten(
+                [
+                    sampled(
+                        make_take_template(
+                            taker,
+                            takee,
+                            use_adverbial_path_modifier=use_adverbial_path_modifier,
+                            spatial_properties=[HARD_FORCE]
+                            if hard_force
+                            else [SOFT_FORCE],
+                        ),
+                        ontology=GAILA_PHASE_1_ONTOLOGY,
+                        chooser=PHASE1_CHOOSER_FACTORY(),
+                        max_to_sample=num_samples,
+                    )
+                    for use_adverbial_path_modifier in BOOL_SET
+                    for hard_force in BOOL_SET
+                ]
+            )
+        ),
+    )
+
+
+def make_push_shove_subtle_verb_distinctions(
+    num_samples: int = 5, *, num_noise_objects: int = 0  # pylint:disable=unused-argument
+) -> Phase1InstanceGroup:
+    pusher = standard_object("pusher_0", THING, required_properties=[ANIMATE])
+    pushee = standard_object("pushee_0", THING, required_properties=[INANIMATE])
+    push_surface = standard_object(
+        "push_surface_0", THING, required_properties=[INANIMATE]
+    )
+    push_goal = standard_object("push_goal_0", THING, required_properties=[INANIMATE])
+    # get all possible templates
+    all_templates = [
+        make_push_templates(
+            pusher,
+            pushee,
+            push_surface,
+            push_goal,
+            use_adverbial_path_modifier=use_adverbial_path_modifier,
+            spatial_properties=[HARD_FORCE] if hard_force else [SOFT_FORCE],
+        )
+        for use_adverbial_path_modifier in BOOL_SET
+        for hard_force in BOOL_SET
+        for express_surface in BOOL_SET
+    ]
+    # format these template pairs into one long list
+    formatted_templates = [template[0] for template in all_templates] + [
+        template[1] for template in all_templates
+    ]
```

Comment:
You can avoid the need to flatten this listOfLists by hand by using `flatten()` from `itertools`

```suggestion
templates = flatten(
        [
            make_push_templates(
                pusher,
                pushee,
                push_surface,
                push_goal,
                use_adverbial_path_modifier=use_adverbial_path_modifier,
                spatial_properties=[HARD_FORCE] if hard_force else [SOFT_FORCE],
            )
            for use_adverbial_path_modifier in BOOL_SET
            for hard_force in BOOL_SET
        ]
    )
```